content/blog/monkey-patch.md

```diff
@@ -0,0 +1,62 @@
++++
+title = "Monkey Patching"
+slug = "monkey-patch"
+date = 2023-10-17T18:29:54+05:30
+image = "/images/2023/10/monkeypatch.jpg"
+draft = false
+authors = ["Nitin Rajkumar"]
+description = "Monkey patching and it's sideeffects in Ruby"
+tags = ["Software Craftsmanship", "GraphQL"]
+categories = ["Software Craftsmanship", "Ruby", "Monkey Patching"]
+type = ""
++++
+
+Ruby, being a dynamic programming language provides the ability and hence freedom to reopen the existing classes and change their behavior during runtime by adding new methods or modifying existing methods. 
+
+#### How to monkey patch
+
+```ruby
+class Sound
+	def car
+		"honk"
+	end
+end
+class Sound
+	def bike
+		"peep-peep"
+	end
+end
+noise = Sound.new
+noise.car #=> honk
+noise.bike #=> peep-peep
+```
+The above is a simple example where we reopened Sound class and added a new method. It provides many other opportunities including overriding the method of the original class.
+
+#### When does one think of using it
+Usually when the programmer wants to fix the bug quickly.
+1. When the gem or library that is being used has a bug and before a patch-fix from the library if one wants to fix it in their code base.
+2. When certain method you expect the library to contain is missing one can monkey patch the method.
+
+#### Side Effects
```

Comment:
Point number 1,2,3,4 on line 50 to 53 all talk about negative effects of using Monkey Patching

The word Side effects does not mean negative Side effects by default

May be the word "Cons" suits better here